dfkernel/ipkernel.py

```diff
@@ -431,6 +476,73 @@ async def run_cell(*args, **kwargs):
 
         return reply_content, res
 
+    def update_code_cells(self, dfmetadata, update_persistent_code=False):
+        curr_output_tags = dict()
```

Comment:
See previous comment about defaultdict, but also please be consistent in the way you are initializing dictionaries the following lines contain mixed usage of {} and dict()

Reply:
Refactored code and used defaultdict in the code